File: app/src/main/java/com/test/xyz/demo/presentation/weather/di/WeatherFragmentModule.java
package com.test.xyz.demo.presentation.weather.di;

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope;
import com.test.xyz.demo.presentation.weather.presenter.WeatherDataFormatter;
import com.test.xyz.demo.presentation.weather.presenter.WeatherPresenter;
import com.test.xyz.demo.presentation.weather.presenter.WeatherPresenterImpl;
import com.test.xyz.demo.presentation.weather.presenter.WeatherView;

import dagger.Module;
import dagger.Provides;

@Module
public class WeatherFragmentModule {
    private final WeatherView view;

    public WeatherFragmentModule(WeatherView view) {
        this.view = view;
    }

    @Provides
    @ActivityScope
    WeatherView provideMainView() {
        return this.view;
    }

    @Provides
    @ActivityScope
    WeatherDataFormatter provideWeatherDataFormatter() {
        return new WeatherDataFormatter();
    }

    @Provides
    @ActivityScope
    WeatherPresenter provideMainPresenter(WeatherPresenterImpl presenter) {
        return presenter;
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/weather/di/WeatherFragmentModule.kt
package com.test.xyz.demo.presentation.weather.di

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope
import com.test.xyz.demo.presentation.weather.presenter.*
import dagger.Module
import dagger.Provides

@Module
class WeatherFragmentModule(private val view: WeatherView) {

    @Provides
    @ActivityScope
    internal fun provideMainView(): WeatherView {
        return this.view
    }

    @Provides
    @ActivityScope
    internal fun provideWeatherDataFormatter(): WeatherDataFormatter {
        return WeatherDataFormatter()
    }

    @Provides
    @ActivityScope
    internal fun provideWeatherDegreeConverterProxy(): WeatherDegreeConverterProxy {
        return WeatherDegreeConverterProxy()
    }

    @Provides
    @ActivityScope
    internal fun provideMainPresenter(presenter: WeatherPresenterImpl): WeatherPresenter {
        return presenter
    }
}
